applications/TrilinosApplication/tests/test_trilinos_linear_solvers.py

```diff
@@ -2,13 +2,24 @@
 import KratosMultiphysics.TrilinosApplication
 import KratosMultiphysics.KratosUnittest as KratosUnittest
 from KratosMultiphysics.TrilinosApplication import trilinos_linear_solver_factory
+from KratosMultiphysics.mpi import DataCommunicatorFactory
 import os
 
 def GetFilePath(fileName):
     return os.path.dirname(os.path.realpath(__file__)) + "/" + fileName
 
 
 class TestLinearSolvers(KratosUnittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        default_data_comm = KratosMultiphysics.Testing.GetDefaultDataCommunicator()
+        cls.data_comm_name = "AllExceptFirst"
+        ranks_sub_comm = [i for i in range(1,default_data_comm.Size())]
+        cls.sub_comm = DataCommunicatorFactory.CreateFromRanksAndRegister(default_data_comm, ranks_sub_comm, cls.data_comm_name)

```

Comment:
explaining more in detail:
lets say we run with 4 MPI-ranks (0-3), then this will create a SubCommunicator that is defined ONLY on ranks 1-3 (but not on 0)
=> when doing any MPI-call with this comm on rank 0, then MPI crashes

=> Thats why we have these functions to check whether it is defined on this rank or not:
https://github.com/KratosMultiphysics/Kratos/blob/9a6659637b014654caaba493846911f0210f7932/kratos/includes/data_communicator.h#L615-L631